### PR TITLE
Implemented wait strategy for UIElements

### PR DESCRIPTION
**Pull request by gioacchinogioffre**
- Added WebDriverWait property to GenericWebDriver
- Added Clear() method to ITypeable interface
- Implemented waits with expected conditions and error handling for Button and TextField classes

### CODE REVIEW

File: 1.Core/UIElements/Drivers/GenericWebDriver.cs [hunk 1]
```diff
@@ -8,6 +9,8 @@ public class GenericWebDriver
     public static IWebDriver Instance =>
         _driver = _driver == null ? WebDriverFactory.GetDriver(ConfigModel.DriverType) : _driver;
 
+    public static WebDriverWait Wait => new WebDriverWait(Instance, TimeSpan.FromSeconds(30));
```
Comment (sebasonti-jalauni) [inline 1.Core/UIElements/Drivers/GenericWebDriver.cs:12]: The value for the explicit wait (in this case 30 seconds) should be obtained from the ConfigBuilder

File: 1.Core/UIElements/Web/TextField.cs [hunk 2]
```diff
@@ -7,9 +10,43 @@ public class TextField : BaseWebElement, ITypeable
     {
         public TextField(string name, Locator locator) : base(name, locator) { }
 
+        public void Clear()
+        {
+            try
+            {
+                GenericWebDriver.Wait.Until(ExpectedConditions.ElementIsVisible(Locator.GetBy()));
```
Comment (sebasonti-jalauni) [inline 1.Core/UIElements/Web/TextField.cs:17]: It is not necesary to wait for the element to be visible if you are waiting for it to be clickable.
You can confirm that by going to the [source code](https://github.com/DotNetSeleniumTools/DotNetSeleniumExtras/blob/master/src/WaitHelpers/ExpectedConditions.cs), check it out. ElementToBeClickable already checks if the element is visible.

**Review by sebasonti-jalauni — CHANGES_REQUESTED** [review]
No big changes requested

File: 1.Core/UIElements/Drivers/GenericWebDriver.cs (hunk 1 above)
Comment (gioacchinogioffre) [reply 1.Core/UIElements/Drivers/GenericWebDriver.cs]: Fixed

File: 1.Core/UIElements/Web/TextField.cs (hunk 2 above)
Comment (gioacchinogioffre) [reply 1.Core/UIElements/Web/TextField.cs]: Cool, I didn't know that thanks. Fixed.

**Review by svaleria — APPROVED** [review]

**Pull request merged**